src/main/java/teammates/storage/api/NotificationsDb.java

```diff
@@ -0,0 +1,144 @@
+package teammates.storage.api;
+
+import static com.googlecode.objectify.ObjectifyService.ofy;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.googlecode.objectify.Key;
+import com.googlecode.objectify.cmd.LoadType;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.entity.Notification;
+
+/**
+ * Handles CRUD operations for notifications.
+ *
+ * @see Notification
+ * @see NotificationAttributes
+ */
+public final class NotificationsDb extends EntitiesDb<Notification, NotificationAttributes> {
+
+    private static final NotificationsDb instance = new NotificationsDb();
+
+    private NotificationsDb() {
+        // prevent initialization
+    }
+
+    public static NotificationsDb inst() {
+        return instance;
+    }
+
+    /**
+     * Gets a notification by its unique ID.
+     */
+    public NotificationAttributes getNotification(String notificationId) {
+        assert notificationId != null;
+
+        return notificationId.isEmpty() ? null : makeAttributesOrNull(getNotificationEntity(notificationId));
+    }
+
+    /**
+     * Gets all notifications.
+     */
+    public List<NotificationAttributes> getAllNotifications() {
+        List<Notification> notifications = load().list();
+        List<NotificationAttributes> notificationAttributes = makeAttributes(notifications);
+        NotificationAttributes.sortByStartTime(notificationAttributes);
+        return notificationAttributes;
+    }
+
+    /**
+     * Gets notifications by {@code targetUser}.
+     *
+     * @return a list of notifications for the specified targetUser.
+     */
+    public List<NotificationAttributes> getActiveNotificationsByTargetUser(NotificationTargetUser targetUser) {
+        assert targetUser != null;
+
+        List<Notification> endEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("endTime >", Instant.now())
+                .list();
+
+        endEntities.addAll(load()
+                .filter("targetUser", NotificationTargetUser.GENERAL)
```

Comment:
Should probably check that `targetUser` is not already `GENERAL` to begin with, otherwise this is a duplicate query